📖 Document policy for workloads on KCP controlled machines (#4237)
kubernetes-sigs/cluster-api: docs/book/src/tasks/kubeadm-control-plane.md

See the section on [Adopting existing machines into KubeadmControlPlane management][adoption]

### Running workloads on control plane machines

> We don't suggest running workloads on control planes, and highly encourage to avoid it unless absolutely necessary,

Reviewer: ```suggestion
We don't suggest running workloads on control planes, and highly encourage avoiding it unless absolutely necessary,
```

> however, in the case the user wants to run non-control plane workloads on control plane machines he/she/it

Reviewer: ```suggestion
however, in the case the user wants to run non-control plane workloads on control plane machines they
```

Reviewer: "they" is a better gender-neutral pronoun IMO

> is ultimately responsible for ensuring the proper functioning of those workloads, given that KCP is not 

Reviewer: ```suggestion
are ultimately responsible for ensuring the proper functioning of those workloads, given that KCP is not 
```

aware of the specific requirements for each type of workload (e.g. preserving quorum, shutdown procedures etc.).

In order to do so, the user could leverage on the same assumption that applies to all the
Cluster API Machines:

- The Kubernetes node hosted on the Machine will be cordoned & drained before removal (with well
  known exceptions like full Cluster deletion).
- The Machine will respect PreDrainDeleteHook and PreTerminateDeleteHook. see the
  [Machine Deletion Phase Hooks proposal](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20200602-machine-deletion-phase-hooks.md)
  for additional details.

<!-- links -->
[adoption]: upgrading-cluster-api-versions.md#adopting-existing-machines-into-kubeadmcontrolplane-management
[upgrades]: upgrading-clusters.md#how-to-upgrade-the-kubernetes-control-plane-version

> 

Reviewer: ```suggestion
```

See the section on [Adopting existing machines into KubeadmControlPlane management][adoption]

### Running workloads on control plane machines

> The user may want to run non-control plane workloads on control plane machines.

Reviewer 2: We should probably mention that we don't suggest to run workloads on control planes, highly encourage to avoid it unless absolutely necessary

Author: Done, PTAL

> In this case, the user is also the ultimate responsible for ensuring the

Reviewer 3: Maybe we can write it like "the user is ultimately responsible for ensuring ..."

proper functioning of those workloads, given that KCP is not aware of the specific
requirements for each type of workload (e.g. preserving quorum, shutdown procedures etc.).

In order to do so, the user could leverage on the same assumption that applies to all the
Cluster API Machines:

- The Kubernetes node hosted on the Machine will be cordoned & drained before removal (with well

>   know exceptions like full Cluster deletion).

Reviewer 3: NIT; "with well **known"** exceptions like full cluster deletion".

> - The Machine will respect PreDrainDeleteHook and PreTerminateDeleteHook. see the

Reviewer 3: Again a minor nit, camel casing is missing in "see the Machine Deletion Phase Hooks proposal"
